swim/suspicion.go

```diff
@@ -33,62 +33,82 @@ type suspect interface {
 	incarnation() int64
 }
 
+type timerstate struct {
```

Comment:
Why not simply embed time.Timer, add the `state` field and continue to call this a timer?